src/cpp/src/lm_encoding.cpp

```diff
@@ -126,7 +126,7 @@ std::pair<EncodedResults, int32_t> get_lm_encoded_results(
                                                 get_active_sequence_groups),
                                  active_sequence_groups.end());
 
-    while (active_sequence_groups.size() > 0) {
```

Comment:
`active_sequence_groups` can be empty even after prompt is processed, why do we need such change?

Reply:
To avoid code duplication.

Reply:
> active_sequence_groups can be empty even after prompt is processed

how about this?

Reply:
> To avoid code duplication

We could create a lambda for it can call in 2 places.